src/main/java/com/bandwidth/sqs/consumer/SqsConsumerManager.java

```diff
@@ -81,8 +104,28 @@ public synchronized void updateAllocatedInFlightRequests(SqsConsumer consumer, U
         }
     }
 
-    public void queueTask(Runnable task) {
-        threadPool.execute(task);
+    public void queueTask(Function<SqsMessage<T>, Completable> task, int priority, SqsConsumer<T> consumer) {
+
+        //queue a task that we want to run on the thread pool. If there are more tasks than threads
+        //the highest priority tasks will run first
+        threadPoolQueue.add(new PriorityTask<>(priority, consumer, (msg) -> {
+            Completable.defer(() -> task.apply(msg))
+                    .doFinally(taskPermits::release).subscribe();
+        }));
+
+        threadPool.execute(() -> {
```

Comment:
Correct. There is a queue of tasks ordered by priority, and each time one is added, there is a task added to the thread pool that runs the highest priority task from the queue.

Without the queue, they would run in the order they were added to the thread pool